src/Controller/Control.js

```diff
@@ -0,0 +1,213 @@
+import { cloneDeep } from 'lodash'
+import rotateMatrix from 'rotate-matrix'
+import {uniqueId, maxBy} from 'lodash'
+
+function isChanged(arr1, arr2) {
+  return arr1.some((value, index) =>
+    value.x !== arr2[index].x || value.y !== arr2[index].y || value.id !== arr2[index].id
+  );
+}
+
+function checkGameOver(cells, size) {
+
+  const matrix = Array.from(new Array(size), () =>
+    Array.from(new Array(size), () => 0),
+  )
+
+  cells.forEach(cell => {
+    matrix[cell.y][cell.x] = cell
+  })
+
+  for (let i = 0; i < matrix.length; i++) {
+    const column = matrix[i];
+    for (let j = 0; j < column.length; j++) {
+      if(matrix[i+1] && matrix[i+1][j].value === column[j].value) {
+        return false;
+      }
+      if(column[j+1] && column[j+1].value === column[j].value) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+function populateField(cells, size) {
+  const occupiedCoords = new Set()
+
+  cells.forEach(cell => {
+    occupiedCoords.add(cell.x * size + cell.y)
+  })
+
+  if (occupiedCoords.size === size**2) return
+
+  let x
+  let y
+  let startSize = occupiedCoords.size
+  do {
+    x = Math.floor(Math.random() * (size-1+0.9))
+    y = Math.floor(Math.random() * (size-1+0.9))
+
+    const sum = x * size + y
+    occupiedCoords.add(sum)
+  } while (startSize === occupiedCoords.size)
+
+  let maxId = Number(maxBy(cells, (cell) => Number(cell.id)).id);
+  let id = uniqueId();
+  while (id <= maxId) {
+    id = uniqueId();
+  }
+
+  return [...cells, create(x, y, (Math.random() - .2 > 0 ? 2 : 4), id)]
+}
+
+function removeAndIncreaseCells (cells) {
+  return cells.filter(cell => cell.state !== cellStates.DYING).map(cell => {
+    if (cell.state === cellStates.INCREASE) {
+      cell.value *= 2
+    }
+
+    cell.state = cellStates.IDLE
+
+    return cell
+  })
+}
+
+const directions = {
```

Comment:
If it's an object with readonly properties then you could name it in uppercase (like `DIRECTIONS`) and move it at the top of the file